### ISSUE TITLE
Archive file selector

### ISSUE THREAD

**Issue by fcollonval** [hadim/jupyter-archive#19]
Enhancement

Use a file selector to show the context action only on archive files.

The filename is stored as title attribute in the file browser. Using CSS selector with attribute filtering maybe a solution:

https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

**Comment by hadim** [comment]
I tried `const selectorOnlyArchive = '.jp-DirListing-item[title$=".zip"], .jp-DirListing-item[title$=".tgz"]';` but it does not work.

A possibly related issue: https://github.com/phosphorjs/phosphor/issues/354

**Comment by hadim** [comment]
Before we find a more elegant solution, the following does the job:

```js
// Add a command for each archive extensions
    // TODO: use only one command and accept multiple extensions.
    const allowedArchiveExtensions = [".zip", ".tar.gz"]

    allowedArchiveExtensions.forEach(extension => {
      const selector = '.jp-DirListing-item[title$="' + extension + '"]';
      app.contextMenu.addItem({
        command: CommandIDs.extract_archive,
        selector: selector,
        rank: 10
      });
    });
```

**Comment by hadim** [comment]
Closed by 7535f1289e37032d60ab61e3f3b05ed07dc610f5
